dataless-classifier/src/main/java/edu/illinois/cs/cogcomp/datalessclassification/representation/AEmbedding.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * This software is released under the University of Illinois/Research and Academic Use License. See
+ * the LICENSE file in the root folder for details. Copyright (c) 2016
+ *
+ * Developed by: The Cognitive Computation Group University of Illinois at Urbana-Champaign
+ * http://cogcomp.cs.illinois.edu/
+ */
+package edu.illinois.cs.cogcomp.datalessclassification.representation;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import edu.illinois.cs.cogcomp.datalessclassification.util.DenseVector;
+import edu.illinois.cs.cogcomp.datalessclassification.util.DenseVectorOperations;
+import edu.illinois.cs.cogcomp.datalessclassification.util.QueryPreProcessor;
+import edu.illinois.cs.cogcomp.datalessclassification.util.SparseVector;
+
+/**
+ * @author yqsong@illinois.edu
+ * @author shashank
+ */
+
+public abstract class AEmbedding<T extends Serializable> {
+
+    /**
+     * Override this function to get your desired query segmentation (simpleSegmentation being one
+     * example)
+     */
+    public List<String> getTerms(String query) {
+        return simpleSegmentation(query);
+    }
+
+    final private List<String> simpleSegmentation(String query) {
+        query = QueryPreProcessor.process(query);
+        String[] terms = query.split("\\s+");
+
+        List<String> termList = Arrays.asList(terms);
+
+        return termList;
+    }
+
+    /**
+     * Override this function to get your desired term processing module (simpleProcessTerm being
+     * one example)
+     */
+    public String processTerm(String s) {
+        return simpleProcessTerm(s);
+    }
+
+    final private String simpleProcessTerm(String s) {
+        return s.toLowerCase().trim();
+    }
+
+    /**
+     * Implement this function for returning the default vector in case no word in the query could
+     * be found in the index
+     * 
+     * @throws java.io.IOException
+     */
+    abstract public SparseVector<T> getDefaultConceptVectorMap();
+
+    /**
+     * Intended to output the vector for the most basic unit Should internally process the term
+     * using the processTerm function
+     * 
+     * @throws java.io.IOException
+     */
+    abstract public SparseVector<T> getTermConceptVectorMap(String term);
+
+    /**
+     * Intended to output the vector for any arbitrary query
+     * 
+     * @throws java.io.IOException
+     */
+    abstract public SparseVector<T> getVector(String query);
+
+    /**
+     * Override this function to select given number of dimensions
+     */
+    public SparseVector<T> getVector(String query, int numConcepts) {
+        return getVectorIgnoreSize(query);
+    }
+
+    private final SparseVector<T> getVectorIgnoreSize(String query) {
+        return getVector(query);
+    }
+
+    public SparseVector<T> getConceptVectorBasedonSegmentation(String query) {
```

Comment:
javadoc explaining what 'segmentation' refers to
nitpick: capitalize the 'O' in 'basedOn'

Reply:
Done.